### ISSUE TITLE
Request Swiss German

### ISSUE THREAD

**Issue by yagan93** [common-voice/common-voice#3034]
Use this template to request a new localizable language that is currently not available [on Pontoon](https://pontoon.mozilla.org/projects/common-voice/).

Note: This issue only applies to the web interface language - in order to activate language contributions on Common Voice you will also need to ensure 5,000+ sentences are available to be read in that language. Please refer to the [full Language documentation](https://github.com/mozilla/common-voice/blob/master/README.md) for more details.

**Language name**
Swiss German

**Language code**
gsw

**Language size**
4'900'000

**Plural forms**
How would you translate the following in this language?

0 Stei
1 Stei 
2 Stei 
3 Stei 
4 Stei 
5 Stei 
10 Stei
20 Stei
100 Stei
1000 Stei
Ich gsehn 0 Stei ufem Bode
Ich gsehn 1 Stei ufem Bode
Ich gsehn 10 Stei ufem Bode
Ich gsehn Stei ufem Bode

**Pontoon manager**
https://pontoon.mozilla.org/contributors/mUl2rItJe7HFf9pxMao8BO3rM3k/

**Comment by stefangrotz** [comment]
Hi @yagan93 I worked a lot for the German version of CV and I am from the Alemannic parts of Germany close to the Swiss boarders,  so I also would like to see this language version. But there are a few things that will make it very dificullt:

The main problem is: Swiss German is not standardized, so creating a coherent sentence corpus will be hard. The best source of sentences that I have found is the alemanic wikipedia: https://als.wikipedia.org/wiki/Wikipedia:Houptsyte We could extract maybe around 60 000 sentences from there if we are lucky with the wiki extractor.

Honestly: I can only imagine two possible ways. Either you create some sort of standardized swiss german or you select a specific dialect from a swiss region such a Berndeutsch. The third possibility would be to accept the diversity of the dialects. But this means that people will have to read sentences, that they are not used to.

**Comment by yagan93** [comment]
Hi @stefangrotz. I am currently elaborating on wav2vec 2.0 - XLSR-53 (https://github.com/pytorch/fairseq/blob/master/examples/wav2vec/README.md) released by Facebook last year. Im trying to find out if we could successfully combine swiss german as a low resource language with german as a high resource language, thus focusing more on a swiss german to german transcription. The only labeled corpus that is available at the moment was published by FHNW (https://drive.switch.ch/index.php/s/cOLa8sp2rKpf4s0), which is of semi good quality. Therefore I thought of fine tuning the trained model with pseudo labeling (maybe even iteratively). At the moment  I am trying to get a better understanding of how the internas of XLSR really work and how the published pretrained model behaves in real world scenarios. Im still having some issues to let it run as it is not yet available through the transformers API / huggingface.co. Maybe we could team up and see what we can achieve as a team? :)

**Comment by stefangrotz** [comment]
Interesting. So you want to create a system that transcribes spoken Swiss German to written High German, right? 
Common Voice doesn't work like this, the spoken language has to correspond directly to the written language, only accents are accepted, but no variations in the words and how people express things. The goal of Common Voice is to create a dataset that is optimized for the [Deepspeech](https://github.com/mozilla/DeepSpeech) neural network. 

What kind of data do you expect to get from the Swiss German Common Voice dataset?  Do you want written Dialect or should the sentences be written in High German?

If you are looking for data with a Swiss accent, 1% of the [german dataset](https://commonvoice.mozilla.org/de/datasets) are labeled Swiss German, you can filter them and get around 80 hours of High German with a Swiss accent.  If you want this kind of data the easiest way to get more would be to promote the German version of Common Voice and motivate people to create a profile and select Swiss German as their accent.

**Comment by Heyhillary** [comment]
Hey @yagan93,

Thanks for your request - I'm messgaing to follow up on your request. You might be intrested to learn about how we are considering including variants of langauges for 2022. Please read this blog to learn more: https://foundation.mozilla.org/en/blog/how-we-are-making-common-voice-even-more-linguistically-inclusive/

**Comment by stefangrotz** [comment]
Interesting, maybe this should even become a completely new issue. One could add variants such as Namibian German, High German with Swiss orthography and numerous dialects from all over Switzerland, Austria and Germany.  

In this issue we should stick to questions related to Swiss German. I could think of two different posibillities:

1. Add the Swiss variant of High German (with orthographic specialities like no ß and a few words that are only used in Switzerland such as velo, spital,... This would be relatively easy
2. Add Swiss dialects. This would be a huge undertaking since there are hundreds of variants and no fixed orthographic rules.

**Comment by ftyers** [comment]
I think that option (1) is a good start, on the other hand, for the sentence-level stuff we would need to wait as the variants will only be supported on clips for now (as far as I understand).

**Comment by MichaelKohler** [comment]
> @MichaelKohler you are from Switzerland, if I remember correctly, right? Do you think a variety of "swiss high german" would be useful?

I mean, sure why not.  There certainly are a few words that are used in the Swiss variant of High German that are not used in "normal" High German. That Wikipedia article is quite interesting. That might indeed be useful once sentences can be tagged accordingly. That work still needs to be scoped and defined though, so not gonna happen within the next weeks.

As for the actual Swiss German, I'm still very much torn whether it's a language or variant as per the blog post. Without having deeper knowledge around linguistics, I probably would argue that it should be its own language, potentially with the different "dialects" as variants. But then I'm not sure how useful that will be overall.

**Comment by Heyhillary** [comment]
Hey everyone, 

Sorry for my silence on this issue.

I have had another person request the inclusion of Swiss German as a language.

In the most recent [weekly update ](https://discourse.mozilla.org/t/weekly-update-our-voice-competition/99130)we have started the second round of community consultations for the additions of language variants.

The recording will be made available but you can [access the slides ](https://docs.google.com/presentation/d/18rafFef5uliLng9VylhHwHAr4m4nTA1fe2L7rL1S6CQ/edit?usp=sharing).

The next deadline to add variants is the end of July (future deadlines in slides).

Would yall be interested in us having a catch-up or async on what the variants could look like for German?

**Comment by stefangrotz** [comment]
I agree with Michael and believe that since Swiss German dialects are not standardized in any way, the only language that could be added relatively easy would be Swiss Standard German. This indeed could help a lot to train models specialized on the accent and the Swiss orthography, but it could also be a great test dataset for German models.

**Comment by ftyers** [comment]
It's worth noting that `gsw` for Swiss German also includes Alsatian, so it would be good to get feedback from speakers of that too.

**Comment by stefangrotz** [comment]
This certainly would be interesting. However, I believe focussing on practical questions is also important.

The main question for me is: **Can we collect enough sentences in `gsw` that are pronounceable for all people across the dialect spectrum?** My understanding is that the dialect differs a lot in different valleys inside Switzerland. Can we expect that a Swiss German would be able and willing to record sentences written in an Alsatian dialect?

Using `de-CH` would be a lot easier in this regard since there are tons of standardized texts available, but it would be a completely different project.

**Comment by MichaelKohler** [comment]
> Can we expect that a Swiss German would be able and willing to record sentences written in an Alsatian dialect?

If the requirement is to pronounce it as it should be in an Alsatian dialect, then no. This can very quickly become a guessing game. If I was allowed to pronounce it in my own dialect, then it would require some translation in my head and in many cases that would work out. But then that would completely defeat the purpose as it's variant overlapping. I don't think that would be the goal, right?

Me, personally, would be annoyed to pretend to speak another dialect than I actually do, and I most likely also wouldn't do a good job at it.

> Do we want a speech assistant that is able to understand and write in dialects, or do we want a speech assistant that can handle Swiss German accents and write in the official orthography? 

Thinking more about this, I don't think this provides much value if it's Swiss German accepts speaking German, as we can cover that within the German as is. For the official orthography I'm pretty sure that could be handled outside of the corpus with transformations on the text itself. And for the few additional words, not sure if that is even worth it.

**Comment by Heyhillary** [comment]
Heya, I'm also reaching out to colleagues at the [VA-Peper project ](https://sites.hslu.ch/va-pepr/en/research/)that are assessing the use of voice assistants in Swiss German Homes.

Considering that the next deadline for variant collection is the end of this month. It would be good if we have a decision before 29th July. Is that alright with everyone? If the decision is to extend the discussion I am also open to this too.

**Comment by stefangrotz** [comment]
I am willing to help with both projects on a low but permanent level. But I do not speak Swiss German, just a close dialect from across the boarder, so I should not have a strong voice in this decision.

I actually believe that for polycentric languages like German some kind of "sub-corpus" inside of the German dataset would make more sense. It would be great if one could choose additional sentences for your region to record and to validate.

**Comment by ftyers** [comment]
This has been stale for some time. If there is another request, we can come back to it.
